### PR TITLE
gh webhook signature unit test

### LINKED ISSUE #78: [Enhancement]: Basic unit tests
### My idea

Create Django's tests for some more complex features like webhooks parsing.

### PR DESCRIPTION
**Pull request by wojpo**
tests for checking if signature is correct

### CODE REVIEW

**Review by Sanielia — DISMISSED** [review]
enchancement -> enhancement,
use kebab-case for branch names,
name PRs in such convention:
<PR_type>: PR_title
So in this case:
Feat: GH webhook signature unit test

File: core/tests.py [hunk 1]
```diff
@@ -1 +1,35 @@
-# Create your tests here.
+import json
+from django.test import TestCase
+import hmac
+import hashlib
+
+from core.gh_webhook_handling import verify_secret
+
+
+def generate_signature(secret, payload):
```
Comment (Norbiros) [inline core/tests.py]: Maybe move this function to `gh_webhook_handling.py`?

**Review by Norbiros — CHANGES_REQUESTED** [review]

File: core/tests.py (hunk 1 above)
Comment (Sanielia) [reply core/tests.py]: `verify_secret` atm more or less containts this function
image

Comment (Norbiros) [reply core/tests.py]: Yes, so now we duplicate code. So to fix it he could “extract” `generate_signature` from `verify_secret` method

Comment (wojpo) [reply core/tests.py]: I changed this looks good rn

**Review by Norbiros — APPROVED** [review]

**Review by Sanielia — APPROVED** [review]

**Pull request merged**